### PR TITLE
Bump rails to 7.1.3

### PR DESCRIPTION
**Pull request by ryan-mcneil**
## Summary

- Bump to Rails 7.1.X (Currently 7.1.3.2)
- No notable changes, just the deprecation of a UUID setting (see linked PRs below), and the introduction of several new deprecation warnings to be handled after merging and testing this PR.
- A few very minor adjustments were required, and noted in PR

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/73307

### Also completes
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/76187

### Related PR's ahead of this update
- https://github.com/department-of-veterans-affairs/vets-api/pull/15295
- https://github.com/department-of-veterans-affairs/vets-api/pull/15219
- https://github.com/department-of-veterans-affairs/vets-api/pull/15917

### PR to follow this one (handle deprecation warnings)
- https://github.com/department-of-veterans-affairs/vets-api/pull/15919 (must pull in master after merge)

## Testing done

- Passing Test Suite
- Standby for production issues.

## What areas of the site does it impact?
* Certain syntax changes will be enforced moving forward by the new version

### CODE REVIEW

File: modules/test_user_dashboard/app/models/test_user_dashboard/tud_account.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ class TudAccount < ApplicationRecord
     validates :email, uniqueness: true
     validate :valid_id_types
 
-    serialize :services
+    serialize :services, coder: YAML
```
Comment (ryan-mcneil) [inline modules/test_user_dashboard/app/models/test_user_dashboard/tud_account.rb:13]: YAML was [the default prior to 7.1](https://guides.rubyonrails.org/configuring.html#config-active-record-default-column-serializer), so I restored it here. However, maybe we should use something else? Can we use the same coder as everywhere else (`JsonMarshal::Marshaller`)?

File: modules/check_in/spec/services/map/service_spec.rb [hunk 2]
```diff
@@ -41,7 +41,7 @@
       end
 
       it 'returns token from redis' do
-        expect(subject.build.token).to eq(access_token)
+        expect(subject.build(opts).token).to eq(access_token)
```
Comment (ryan-mcneil) [inline modules/check_in/spec/services/map/service_spec.rb:44]: @kanchanasuriya Hi, can you confirm this addition is okay and doesn't change the intent of the test?

If `opts` is intended to be left out, we need to come up with a code change to properly handle a change in Rials 7.1. According to the docs they're now raising an error if a key isn't passed to the `ActiveStorage::Cache` ([Changelog](https://github.com/rails/rails/blob/7-1-stable/activesupport/CHANGELOG.md?plain=1#L462): "Consistently raise an ArgumentError if the ActiveSupport::Cache key is blank.").

Depending on what the expected behavior is, we may need to rescue this error in the Map::Service. Let me know your thoughts and we can chat about it. Thanks!

Comment (kanchanasuriya) [reply modules/check_in/spec/services/map/service_spec.rb:44]: Changes look good, @ryan-mcneil.  Method response is anyway mocked and adding this parameter doesn't change any behaviour in the test.

**Review by kanchanasuriya — APPROVED** [review]
LGTM

File: modules/apps_api/config/locales/en.yml [hunk 3]
```diff
@@ -9,7 +9,7 @@ en:
       app_type: 'The type of the application.'
       description: 'A baseline description of the application including its goals and a short summary of the history of the application.'
       platforms: 'The platform(s) this application is available on.'
-      icon_url: 'The URL of the logo for the given application.'
+      logo_url: 'The URL of the logo for the given application.'
```
Comment (ryan-mcneil) [inline modules/apps_api/config/locales/en.yml:12]: Somehow this has always been wrong, but just started throwing errors in the new version

File: Gemfile.lock [hunk 4]
```diff
@@ -1038,8 +1072,9 @@ GEM
     zeitwerk (2.6.13)
 
 PLATFORMS
+  arm64-darwin-22
```
Comment (stevenjcumming) [inline Gemfile.lock:1075]: https://github.com/department-of-veterans-affairs/vets-api/pull/15143#issuecomment-1984711913

bundle lock --add-platform ruby
bundle lock --remove-platform arm64-darwin-22
bundle install --redownload

Comment (ericboehs) [inline Gemfile.lock:1075]: This needs to be switched back to `ruby`.

Comment (stevenjcumming) [reply Gemfile.lock:1075]:
```
bundle lock --add-platform ruby
bundle lock --remove-platform arm64-darwin-22
bundle install --redownload
```

File: modules/check_in/spec/services/check_in/map/token_service_spec.rb [hunk 5]
```diff
@@ -41,7 +41,7 @@
       end
 
       it 'returns token from redis' do
-        expect(subject.build.token).to eq(access_token)
+        expect(subject.build(opts).token).to eq(access_token)
```
Comment (ryan-mcneil) [inline modules/check_in/spec/services/check_in/map/token_service_spec.rb:44]: @kanchanasuriya previously confirmed these changes to the expectation were ok 👌

File: Gemfile.lock [hunk 6]
```diff
@@ -1038,6 +1072,7 @@ GEM
     zeitwerk (2.6.13)
 
 PLATFORMS
+  arm64-darwin-23
```
Comment (ericboehs) [inline Gemfile.lock:1075]: Still shouldn't have this afaik.

File: modules/test_user_dashboard/app/models/test_user_dashboard/tud_account.rb (hunk 1 above)
Comment (ryan-mcneil) [reply modules/test_user_dashboard/app/models/test_user_dashboard/tud_account.rb]: This will be moved to a followup PR

File: Gemfile.lock (hunk 6 above)
Comment (ryan-mcneil) [reply Gemfile.lock:1075]: wait wtf

Comment (ryan-mcneil) [reply Gemfile.lock:1075]: oh both `arm64-darwin-22` and `arm64-darwin-23` were in there. weird. 💣

**Review by kpethtel — DISMISSED** [review]

File: modules/test_user_dashboard/app/models/test_user_dashboard/tud_account.rb (hunk 1 above)
Comment (ryan-mcneil) [reply modules/test_user_dashboard/app/models/test_user_dashboard/tud_account.rb:13]: @bramleyjl Perhaps you can provide a suggestion for how we move forward here? Now that something must be specified, do we use the pre-7.1 default (`YAML`) or would it make sense to be consistent with the rest of the app?

**Review by stevenjcumming — APPROVED** [review]

**Review by stevenjcumming — APPROVED** [review]

**Pull request merged**